core/client/fs/src/main/java/alluxio/client/block/policy/CapacityBaseRandomPolicy.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.block.policy;
+
+import alluxio.client.block.BlockWorkerInfo;
+import alluxio.client.block.policy.options.GetWorkerOptions;
+import alluxio.wire.WorkerNetAddress;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Random load block base on capacity of worker,
+ * used for cluster of workers with different capacity.
+ * Then larger capacity worker will receive more load request.
+ *
+ */
+@ThreadSafe
+public class CapacityBaseRandomPolicy implements BlockLocationPolicy {
+
+  /**
+   * used for binary search.
+   */
+  private static final Comparator<CapacityRangeWorkerInfo> SEARCH_COMPARATOR = (o1, o2) -> {
+    if (o1.mStartInclusive <= o2.mStartInclusive && o1.mEndExclusive >= o2.mEndExclusive) {
```

Comment:
Looks like it returns zero when the range of o1 equals or includes the range of o2,  but how about the range of o2 includes the range of o1?

Reply:
I see, I think the range of o2 is just one, but you won't put 0 capacity worker into the list.  But it looks like too magic to me.
```
binarySearch(workerList, new CapacityRangeWorkerInfo(randomLong, randomLong + 1, null),
```